### PR TITLE
Add `EndpointId` to IR

### PR DESCRIPTION
**Pull request by amckinney**
This adds a `commons.EndpointId` alias and includes it in the `HttpEndpoint` type. A `generateEndpointId` method already existed for the audience graph, so this reuses the same form already defined there.

For now, the change remains as simple as possible and includes it in the pre-existing `IRv26`, but we might later want to create another top-level `map<EndpointId, HttpEndpoint` map so that generators can quickly access all of the endpoints without first traversing the services. We might also consolidate the `EndpointName` and `EndpointId` in a single type (similar to the `DeclaredTypeName` object).

### CODE REVIEW

File: packages/cli/generation/ir-generator/src/IdGenerator.ts [hunk 1]
```diff
@@ -24,9 +18,9 @@ export const IdGenerator = {
         const joinedFernFilePath = stringifyFernFilepath(fernFilepath);
         return `subpackage_${joinedFernFilePath}`;
     },
-    generateEndpointId: (declaredServiceName: DeclaredServiceName, httpEndpoint: HttpEndpoint): string => {
+    generateEndpointId: (declaredServiceName: DeclaredServiceName, httpEndpointName: Name): string => {
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/IdGenerator.ts:21]: can we pass the `HttpEndpoint` directly, a lot of things have `Name` so doesnt provide as much type safety

**Review by dsinghvi — APPROVED** [review]
LGTM, just one small change

File: packages/cli/generation/ir-generator/src/IdGenerator.ts (hunk 1 above)
Comment (amckinney) [reply packages/cli/generation/ir-generator/src/IdGenerator.ts:21]: I originally wanted to, but we actually use this function while we're constructing the `HttpEndpoint` itself, so it's actually easier to use the name here.

See `convertHttpService.ts` for reference:
```typescript
            const endpointName = file.casingsGenerator.generateName(endpointKey);
            return {
                ...convertDeclaration(endpoint),
                name: file.casingsGenerator.generateName(endpointKey),
                id: IdGenerator.generateEndpointId(serviceName, endpointName),
                name: endpointName,
                displayName: endpoint["display-name"],
                auth: endpoint.auth ?? serviceDefinition.auth,
                baseUrl: endpoint.url ?? serviceDefinition.url,
```

Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/IdGenerator.ts:21]: got it you could make the type `Omit<HttpEndpoint, "id">` ? not the cleanest but i think better than name?

**Pull request merged**